cli/src/complete.rs

```diff
@@ -241,6 +241,8 @@ fn revisions(revisions: Option<&str>) -> Vec<CompletionCandidate> {
         const REMOTE_BOOKMARK: usize = 5;
         const REVSET_ALIAS: usize = 6;
 
+        let match_pattern = (!match_prefix.is_empty())
+            .then(|| format!("glob:{}*", glob::Pattern::escape(match_prefix)));
```

Comment:
nit: update `tag list` to do `.filter(..)` consistently, and remove `match_pattern`